In [1]:
import pandas as pd
import numpy as np

In [2]:
# Create arrays
time = np.array([0,1.85,2.87,3.78,4.65, 5.50, 6.32,7.14,7.96,8.79,9.69])
position = np.array([0,10,20,30,40,50,60,70,80,90,100])

In [3]:
# Create dataframe
bolt_data = {"bolt_time":time, "bolt_position":position}
bolt_df = pd.DataFrame(data = bolt_data)

In [4]:
# Check columns
bolt_df.columns.values.tolist()

['bolt_time', 'bolt_position']

In [5]:
from sklearn.linear_model import LinearRegression

In [6]:
# build classifier (clf)
clf = LinearRegression()

# fit linear model
clf.fit(time.reshape(-1,1), position.reshape(-1,1))

LinearRegression()

In [7]:
# tells the weights on the respective features
clf.coef_

array([[10.90064661]])

In [8]:
clf.intercept_

array([-8.02116903])

the right linear regression model for this data set is y = 10.9006x + -8.0211.

For the 2009 Usain Bolt world-record breaking run, the position is predicted by: 10.9006 * time - 8.0211.